examples/checkpointing/checkpoint.py

```diff
@@ -62,27 +63,31 @@ def __init__(self, objects):
     def dtype(self):
         return self._dtype
 
-    def save(self, ptr):
-        """Overwrite live-data in this Checkpoint object with data found at
-        the ptr location."""
-        i_ptr_lo = 0
-        i_ptr_hi = 0
-        for o in self.objects:
-            i_ptr_hi = i_ptr_hi + o.size
-            ptr[i_ptr_lo:i_ptr_hi] = o.data.flatten()[:]
-            i_ptr_lo = i_ptr_hi
-
-    def load(self, ptr):
-        """Copy live-data from this Checkpoint object into the memory given by
-        the ptr."""
-        i_ptr_lo = 0
-        i_ptr_hi = 0
-        for o in self.objects:
-            i_ptr_hi = i_ptr_hi + o.size
-            o.data[:] = ptr[i_ptr_lo:i_ptr_hi].reshape(o.shape)
-            i_ptr_lo = i_ptr_hi
+    def get_data(self, timestep):
+        data = flatten([get_symbol_data(s, timestep) for s in self.objects])
+        return data
+
+    def get_data_location(self, timestep):
+        return self.get_data(timestep)
 
     @property
     def size(self):
         """The memory consumption of the data contained in a checkpoint."""
-        return sum([o.size for o in self.objects])
+        return sum([o.size*o.time_order for o in self.objects])
+
+    def save(*args):
+        raise RuntimeError("Invalid method called. Did you check your version" +
+                           " of pyrevolve?")
+
+    def load(*args):
+        raise RuntimeError("Invalid method called. Did you check your version" +
+                           " of pyrevolve?")
+
+
+def get_symbol_data(symbol, timestep):
```

Comment:
I think "symbol" should actually be "function" ?

Reply:
Why?

Reply:
convention throughout devito. Functions have data attached (Function, TimeFunction, SparseFunction ...). Symbols are more generic . But I guess here you're looking for {...}Functions?